crates/floresta-chain/src/pruned_utreexo/udata.rs

```diff
@@ -139,10 +142,11 @@ pub mod proof_util {
         let proof = Proof::new(targets, hashes);
         let mut hashes = Vec::new();
         let mut leaves_iter = udata.leaves.iter().cloned();
-        let mut tx_iter = transactions.iter();
+
+        let tx_iter = block.txdata.iter();
 
         let mut inputs = HashMap::new();
-        tx_iter.next(); // Skip coinbase
+        let actual_block_height = block.bip34_block_height().expect("Bip34 Height"); // Skip coinbase
 
```

Comment:
Including heigh extracted from the block and spent utxos now include commited height

Reply:
Don't use bip34 height. BIP034 was enabled around block 200k, so all blocks before that lack this value. Furthermore, we can't really trust it. Just use `chain.get_block_height`.